src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphMethodConfigurationDAO.scala

```diff
@@ -0,0 +1,117 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import com.tinkerpop.blueprints.{Graph, Vertex, Direction}
+import com.tinkerpop.gremlin.java.GremlinPipeline
+import org.broadinstitute.dsde.rawls.model.{MethodConfigurationShort, Method, WorkspaceName, MethodConfiguration}
+import scala.collection.JavaConversions._
+
+/**
+ * Graph implementation of method config dao. Method configs are stored as a
+ * top level vertex and 3 subordinate vertices for each of inputs, outputs and
+ * prerequisites. Which sub-vertex is which is recorded on the connecting edge.
+ */
+class GraphMethodConfigurationDAO extends MethodConfigurationDAO with GraphDAO {
+  private val InputsEdge: String = "inputs"
+  private val OutputsEdge: String = "outputs"
+  private val PrerequisitesEdge: String = "prerequisites"
+
+  /** gets by method config name */
+  override def get(workspaceNamespace: String, workspaceName: String, methodConfigurationNamespace: String, methodConfigurationName: String, txn: RawlsTransaction): Option[MethodConfiguration] = txn withGraph { graph =>
+    // notice that methodConfigPipe is a def, not a val so we get a new pipeline every time
+    def methodConfigPipe = methodConfigPipeline(graph, workspaceNamespace, workspaceName, methodConfigurationNamespace, methodConfigurationName)
+
+    getSinglePipelineResult(methodConfigPipe.map()).asInstanceOf[Option[java.util.Map[String, String]]] map { methodConfigProps =>
```

Comment:
Also I generally dislike using GremlinPipeline's ```map```, since it doesn't do the same thing as Scala's ```map```, so it's a bit confusing to read and doesn't really seem intuitive in the context of a pipeline. But that's just personal opinion.

Perhaps we could define a custom function in ```GraphDAO``` called ```getPropertyMap``` or something, which gets you the property map for a given Gremlin pipeline endpoint? That would make things clearer, and would also save us the trouble of having ```asInstanceOf[Option[java.util.Map[String, String]]]``` everywhere.